rotkehlchen/chain/evm/decoding/thegraph/balances.py

```diff
@@ -191,13 +166,30 @@ def query_balances(self) -> 'BalancesSheetType':
             )[0]
 
             # calculate current tokens balance relative to the pool state and shares distribution
-            if pool_total_shares == 0:
+            if pool_total_shares == ZERO:
                 continue
             balance = FVal(shares_amount * pool_total_tokens / pool_total_shares)
             balance_norm = asset_normalized_value(balance.to_int(exact=False), self.token)
-            balances[delegator].assets[self.token] += Balance(
-                amount=balance_norm,
-                usd_value=grt_price * balance_norm,
-            )
+            if balance_norm > ZERO:
+                balances[user_address].assets[self.token] += Balance(
+                    amount=balance_norm,
+                    usd_value=grt_price * balance_norm,
+                )
 
         return balances
+
+    def query_balances(self) -> BalancesSheetType:
+        """Queries and returns the balances sheet for staking events.
+
+        Retrieves deposit events and processes them to generate a unique set of delegations.
+        Supports both simple and vested staking."""
+        balances: BalancesSheetType = defaultdict(BalanceSheet)
+        # On Arbitrum, we also want process Ethereum HistoryEvents, cause of the access to
+        # required extra_data (delegator_l2, indexer_l2)
+        location = Location.ETHEREUM if self.evm_inquirer.chain_id == ChainID.ETHEREUM else None
+        events = self._get_staking_events(location)  # fetch deposit events
```

Comment:
So this is close but not exactly what I asked.

See here you are having a conditional location based on the evm_inqurier.chain_id. But I said to give it as an argument. This is how inheritance works. You don't check in the inherited function what the class is.

1. So rename this to something like _base_balance_query(location: Location)
2. Implement query_balances for both arbitrum and ethereum and the implementation should be just calling this base function with a different argument. This way you save this conditional check in each call
3. Nitpick: Can also use walrus operator more. For example:

```python
if (events := self._get_staking_events(location)) is None:
    return balances
```